src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,63 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+
+    private static final int MAX_VALUE = 5;
+    private static final Random random = new Random();
+    private final ColorSupplier colorSupplier;
+
+    public FigureSupplier(ColorSupplier colorSupplier) {
+        this.colorSupplier = colorSupplier;
+    }
+
+    public static Figure getDefaultFigure() {
+        return new Circle("White", 10);
+    }
+
+    public static Figure getRandomFigure() {
+        int figureType = random.nextInt(MAX_VALUE);
+
+        switch (figureType) {
+
+            case 0:
+                int side = random.nextInt(10) + 1;
+                String colorSquare = ColorSupplier.getRandomColor();
```

Comment:
Don't use static methods in your solution. Instead of calling `ColorSupplier.getRandomColor()` statically, use the instance of `ColorSupplier` provided in the constructor (`colorSupplier`).